server/test/integration/post.spec.js

```diff
@@ -495,6 +495,125 @@ describe('Posts API', () => {
     done();
   });
 
+  it('tests creating and updating a comment', async done => {
+    const postData = {
+      title: 'Test post',
+      body: 'This is the body for a test post',
+    };
+
+    const postUrl = '/api/posts/';
+    const response = await supertest(app)
+      .post(postUrl)
+      .send(postData);
+
+    expect(response.status).toBe(201);
+
+    const createdPost = response.body;
+    const commentUrl = '/api/comments/';
+    const commentData = {
+      body: 'This is the body for a test comment',
+      parentID: createdPost._id,
+    };
+
+    const response1 = await supertest(app)
+      .post(commentUrl)
+      .send(commentData);
+    expect(response1.status).toBe(201);
+
+    // Grab the comment ID with another GET request
+    const response2 = await supertest(app).get(postUrl.concat(createdPost._id));
+
+    // update the comment
+    const updatedComment = {
+      commentBody: 'Second Body',
+      parentID: createdPost._id,
+    };
+
+    const response3 = await supertest(app)
+      .put(commentUrl.concat(response2.body.comments[0]._id))
+      .send(updatedComment);
+    expect(response3.status).toBe(200);
+
+    // validate comment is updated
+    const response4 = await supertest(app).get(postUrl.concat(response.body._id));
+    expect(response4.body.comments[0].body).toBe(updatedComment.commentBody);
+    expect(response4.body.comments[0].parentID).toBe(updatedComment.parentID);
+
+    done();
+  });
+
+  it('tests creating and updating a comment that doesnt exist', async done => {
+    const postData = {
+      title: 'Test post',
+      body: 'This is the body for a test post',
+    };
+
+    const postUrl = '/api/posts/';
+    const response = await supertest(app)
+      .post(postUrl)
+      .send(postData);
+
+    expect(response.status).toBe(201);
+
+    const createdPost = response.body;
+    const commentUrl = '/api/comments/';
+
+    // update the comment
+    const updatedComment = {
+      commentBody: 'Second Body',
+      parentID: createdPost._id,
+    };
+
+    const response3 = await supertest(app)
+      .put(commentUrl)
```

Comment:
This is not actually calling any of our endpoints. This is calling /api/comments (with no id) instead of an invalid id.
You would expect the body to come back with an object
`{ message: 'Cast to ObjectId failed for value "<non_existant_id" at path "_id" for model "Comment" }`, however, this is just going to call a non-existent endpoint, (coincidentally giving the same error code)
Instead, commentUrl  should be '/api/comments/non_existant_id'

Reply:
I've incorporated both changes - this test now gives the message you've mentioned, but validating this explicitly is a bit convoluted (prettier doesn't like it).  
Is it necessary to add this validation, or is the response.status equal to 404 sufficient?

Reply:
Checking status code should be alright, that is also consistent with the rest of the tests.